pwiz_tools/Shared/ProteowizardWrapper/MsDataFileImpl.cs

```diff
@@ -612,19 +612,59 @@ public double[] GetTotalIonCurrent()
             }
             using (var chromatogram = ChromatogramList.chromatogram(0, true))
             {
-                if (chromatogram == null)
+                return chromatogram?.getIntensityArray()?.data.Storage();
+            }
+        }
+
+        public class QcTrace
+        {
+            public QcTrace(Chromatogram c)
+            {
+                Name = c.id;
+                if (Name.ToLowerInvariant().Contains(@"pressure"))
                 {
-                    return null;
+                    MeasuredQuality = "pressure";
+                    IntensityUnits = "psi";
                 }
-                TimeIntensityPairList timeIntensityPairList = new TimeIntensityPairList();
-                chromatogram.getTimeIntensityPairs(ref timeIntensityPairList);
-                double[] intensities = new double[timeIntensityPairList.Count];
-                for (int i = 0; i < intensities.Length; i++)
+                else if (Name.ToLowerInvariant().Contains(@"flow"))
+                {
+                    MeasuredQuality = "volumetric flow rate";

```

Comment:
Will these values ever get shown to the user? If so, they need to end up in .resx files for translation.

Reply:
If this will be used to plot the QcTraces in the GUI, then indeed there will need to be some translation from the precision-and-CV-oriented language I'm using here (which again I plan to convert to an enumerated value) to something more domain-specific (e.g. just 'flow rate' or localized equivalent).